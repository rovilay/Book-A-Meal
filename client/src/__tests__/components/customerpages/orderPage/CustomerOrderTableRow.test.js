import React from 'react';
import { shallow } from 'enzyme';
import { orderOnEdit, orders } from '../../../__mockData__/orderMock';
import CustomerOrderTableRow from '../../../../components/customerpages/orderPage/CustomerOrderTablerow';

const props = {
  item: {
    ...orderOnEdit,
    portion: 2
  },
  showDetails: jest.fn(),
  mealsUrl: orders[0].Meals,
  editOrder: jest.fn(),
  deleteOrder: jest.fn(),
  deleteRow: jest.fn(),
  isEdit: true,
  actions: {
    delete: true,
    info: true,
    edit: true
  },
  mealId: `${orderOnEdit.orderId}`,
  updatePortion: jest.fn(),
  orderedMealsLength: 5
};

describe('CustomerOrderTableRow component test', () => {
  const setup = () => shallow(<CustomerOrderTableRow {...props} />);


  it('should render CustomerOrderTableRow correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should call `cancelOrder`', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn()
    };

    const cancelOrderSpy= jest.spyOn(wrapper.instance(), 'cancelOrder');
    wrapper.instance().cancelOrder(event);
    expect(cancelOrderSpy).toHaveBeenCalled();
    done();
  });


  it('should call `updateOrderPortion`', (done) => {
    const wrapper = setup();

    document.body.innerHTML =
    '<div>' +
    `<input type="number" id="portion-${orderOnEdit.orderId}" value="2" />`
    '</div>';

    const updateOrderPortionSpy= jest.spyOn(wrapper.instance(), 'updateOrderPortion');
    wrapper.instance().updateOrderPortion();
    expect(updateOrderPortionSpy).toHaveBeenCalled();
    expect()

    done();
  });


  it('should call `deleteOrderRow`', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn()
    };

    const deleteOrderRowSpy= jest.spyOn(wrapper.instance(), 'deleteOrderRow');
    wrapper.instance().deleteOrderRow(event);
    expect(deleteOrderRowSpy).toHaveBeenCalled();

    done();
  });


  it('should call `showOrderDetails`', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn()
    };

    const showOrderDetailsSpy= jest.spyOn(wrapper.instance(), 'showOrderDetails');
    wrapper.instance().showOrderDetails(event);
    expect(showOrderDetailsSpy).toHaveBeenCalled();
    done();
  });


  it('should call `editOrders`', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn()
    };

    const editOrdersSpy= jest.spyOn(wrapper.instance(), 'editOrders');
    wrapper.instance().editOrders(event);
    expect(editOrdersSpy).toHaveBeenCalled();

    done();
  });
});
