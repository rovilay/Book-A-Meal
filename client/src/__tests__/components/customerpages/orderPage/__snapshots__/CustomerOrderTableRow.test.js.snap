// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerOrderTableRow component test should render CustomerOrderTableRow correctly 1`] = `
<div
  className="row"
>
  <p
    className="row-item meal-title"
    data-tip={5}
    key="0"
  >
    5
  </p>
  <p
    className="row-item"
    key="1"
  >
    isuti rd
  </p>
  <p
    className="row-item"
    key="2"
  >
    <Component />
    <Component />
    <Component />
    <Component />
  </p>
  <p
    className="row-item"
    key="3"
  >
    1100
  </p>
  <p
    className="row-item input-div"
    key="4"
  >
    <input
      className="portion"
      defaultValue={2}
      id="portion-5"
      min="1"
      name="portion"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </p>
  <p
    className="row-item actions"
  >
    <button
      className="btn-3 box-shadow"
      onClick={[Function]}
    >
      <FontAwesome
        className="trash info-circle"
        name="info-circle"
        size="2x"
      />
    </button>
    <button
      className="btn-3 box-shadow"
      onClick={[Function]}
    >
      <FontAwesome
        className="pencil"
        name="pencil"
        size="2x"
      />
    </button>
    <button
      className="btn-3 box-shadow"
      onClick={[Function]}
    >
      <FontAwesome
        className="trash"
        name="trash"
        size="2x"
      />
    </button>
  </p>
  <ReactTooltip
    className="mytooltip"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
</div>
`;
